Jenkinsfile

```diff
@@ -0,0 +1,133 @@
+            trim: false)
+    }
+    
+    options {
+        buildDiscarder(logRotator(artifactNumToKeepStr: '3', artifactDaysToKeepStr: '5', daysToKeepStr: '5', numToKeepStr: '3'))
+        timeout(time: 1, unit: 'HOURS')
+        timestamps()
+        // ansiColor('xterm')
+    }
+
+    // global tools
+    tools {
+        nodejs 'Node12'
+    }
+
+    // pipeline steps
+    stages {
+        // CHECKOUT git repository 
+        stage('Checkout') {
+            steps {
+                git 'https://github.com/aurcame/mdt'
+            }
+        }
+
+        // BUILD: clean .css files and minimize .js files
+        stage('Build') {
+            parallel {
+                // minimize .js files 
+                stage('JS') {
+                    steps {
+                        sh label: 'minimize JS', script: """
+                        cd ${WORKSPACE}/www/js
+                        uglifyjs --timings init.js -o ../min/custom-min.js
+                        """
+                    }
+                }
+
+                // lint .css files
+                stage('TEST') {
+                    when {
+                        allOf {
+                            branch 'master'
+                            expression {
+                                params.RELEASE_TYPE == 'TEST'
+                            }
+
+                        }
+                }
+
+                    steps {
+                        sh label: 'lint CSS', script: """
+                        cd ${WORKSPACE}/www/css
+                        stylelint style.css"""
+                    }
+                }
+
+
+                // clean .css files 
+                stage('CSS') {
+                    steps {
+                        sh label: 'minimize CSS', script: """
+                        cd ${WORKSPACE}/www/css
+                        cleancss -d style.css > ../min/custom-min.css"""
+                    }
+                }
+                
+                // tar artifact
+                stage('Tar artifact') {
+                    steps {
+                        sh label: 'archive', script: """
+                        cd ${WORKSPACE}/www
+                        tar --exclude='./css' --exclude='./js' -c -z -f ../site-archive-${params.RELEASE_TYPE}-${params.RELEASE_VER}-${BUILD_NUMBER}.tgz ."""
+                    }
+                }
+            }
+        }
+
+        // Archiving the project 
+        stage('Archive') {
+            when {
+                expression {
+
+                    params.RELEASE_TYPE == 'RELEASE'
+                }
+            }
+            steps {
+                archiveArtifacts '*.tgz'
+            }
+        }
+
+        // Deployment
+        stage('Deploymnet') {
+            when {
+                expression {
+
+                    params.RELEASE_TYPE == 'RELEASE'
+                }
+            }
+            steps {
+                sh label: 'set_credentials', script: """
```

Comment:
Instead of using credentials file, you can pass AWS credentials as environment variables, so they will be available to jenkins agent only during runtime
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html